week1/Q2.py

```diff
@@ -0,0 +1,51 @@
+class Node:
+	def __init__(self, value = None, next = None):
+		self.value = value
+		self.next = next
+
+
+class LinkedList:
+	def __init__(self):
+		self.first = None
+		self.last = None
+		self.length = 0
+
+	def clear(self):
+		self.__init__()
+
+	def add(self, x):
+		self.length += 1
+		if self.first == None:
+			self.last = self.first = Node(x, None)
+		else:
+			self.last.next = self.last = Node(x, None)
+
+
+def get_kth_to_last(L, k):
+	L_len = L.length
+	if k >= L_len or k < 0:
+		return False
+
+	el = L.first
+	for i in range(L_len - k - 1):
+		el = el.next
+	return el.value
+
+
+def main():
+	L = LinkedList()
+	inp = input('linkedlist elements input\n')
```

Comment:
If you put a newline at the end of the input prompt, you might as well use a print statement.
For the prompt you can use something like a question mark `"? "` and have more detailed instructions before the first input.